GSA/openacr-editor
# Last modified date and openacr version name

Closes https://github.com/GSA/openacr/issues/269

Summary of changes:
* Added last_modified_date and version to JSON structure where last_modified_date is set to today's date whenever a change is made and version is incremented whenever you load a file otherwise it starts from '1'.
* Add readonly fields to about page to view those values.
* Adjusted the report page to show the various dates in a list.
* Updated and added test assertions for the new fields.

**QA new report**

1. After deployment ('Deploy' action) is successful, open site https://gsa.github.io/openacr-editor/.
2. Click 'Start new report' or 'New report'.
3. Navigate to the About page and add fill in some fields.
4. Confirm the 'Last Modified Date' is set to today's date and version is set to 1.
5. Click the 'View report' or 'Report' tab.
6. Confirm you see a green 'Valid Report' box at the top of the report page with the message 'Your report has passed validation'.
7. Confirm the report includes the report date, last modified date, and version in a list.
8. Download the YAML & HTML report and confirm the HTML and YAML include the new fields.
9. Go back to the 'Overview' page
10. Click 'Start new report' or 'New report'.
11. Click 'Open report'.
12. Upload the downloaded YAML file from step 8.
13. Confirm the 'Last Modified Date' is set to today's date and version is set to 2.

**QA existing report**

1. After deployment ('Deploy' action) is successful, open site https://gsa.github.io/openacr-editor/.
2. Click 'Start new report' or 'New report'.
3. Click 'Open report'.
4. Upload a yaml file from OpenACR that already has 'last modified date' and 'version' set to some values (if needed modify the YAML file to set those values before uploading). Try the 'drupal-9.yaml' or any of the yaml files from https://github.com/GSA/openacr/tree/main/openacr where last modified date is 11/16/2021 and version is 12.
5. Navigate to the About page.
6. Confirm the 'Last Modified Data' is set to the one in the YAML file and the version is incremented (e.g. 12 is now 13).
7. Update the OpenACR in multiple places.
8. Confirm the 'Last Modified Date' is now updated to today's date.
